src/main/java/sirius/biz/jobs/params/Autocompleter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.jobs.params;
+
+import sirius.db.mixing.Entity;
+import sirius.kernel.di.std.AutoRegister;
+import sirius.kernel.di.std.Named;
+import sirius.web.controller.AutocompleteHelper;
+import sirius.web.http.WebContext;
+
+import java.util.Collection;
+import java.util.function.Consumer;
+
+/**
+ * An abstract class that can be implemented to supply {@link AutocompleteHelper.Completion Completions} for entity-based token-autocomplete fields.
+ *
+ * @param <T> the type of the entity to autocomplete
+ * @implNote the Autocompleter could be more generic to allow non-entity-autocompletes, but for now, this is not needed.
+ */
+@AutoRegister
+public interface Autocompleter<T extends Entity> extends Named {
+    /**
+     * Computes the possible completions for the query and the web context.
+     * <p>
+     * Implementations may return the items in an ordered collection. This order is retained.
+     *
+     * @param query   the query to search for
+     * @param context the web context with possibly additional context to the search
+     * @return a potentially ordered collection of completions
+     */
+    Collection<T> getCompletions(String query, WebContext context);
```

Comment:
Optional: Rename to `webContext`. `context` is better than `ctx` but as we have quite a view different context classes in sirius in my opinion it helps quite a lot to name them as such.